vortex-scalar/src/binary.rs

```diff
@@ -52,47 +52,58 @@ impl<'a> TryFrom<&'a Scalar> for Buffer {
     type Error = VortexError;
 
     fn try_from(scalar: &'a Scalar) -> VortexResult<Self> {
-        Buffer::try_from(scalar.value())
+        <Option<Buffer>>::try_from(scalar)?
```

Comment:
I think this might be more comprehensible as `scalar.as_binary_opt().ok_or_else(...)`

Reply:
Done.